clang/test/CodeGenSYCL/free_function_int_header.cpp

```diff
@@ -0,0 +1,35 @@
+// RUN: %clang_cc1 -fsycl-is-device -internal-isystem %S/Inputs -triple spir64-unknown-unknown -sycl-std=2020 -fsycl-int-header=%t.h %s -emit-llvm -o %t.ll
+// RUN: FileCheck -input-file=%t.h %s
+// 
+// This test checks integration header contents for free functions with scalar,
+// pointer, simple struct parameters.
+
+#include "mock_properties.hpp"
+#include "sycl.hpp"
+
+struct Simple {
+  int x;
+  char c[100];
+  float f;
+};
+
+__attribute__((sycl_device))
+[[__sycl_detail__::add_ir_attributes_function("sycl-single-task-kernel", 2)]]
+void ff_2(int *ptr, int start, int end, struct Simple S) {
+  for (int i = start; i <= end; i++)
+    ptr[i] = start + S.x + S.f + S.c[2] + 66;
+}
+// CHECK:      const char* const kernel_names[] = {
+// CHECK-NEXT:   "__free_function_ff_2"
+// CHECK-NEXT: };
+
+
+// CHECK:      const kernel_param_desc_t kernel_signatures[] = {
+// CHECK-NEXT:   //--- __free_function_ff_2
```

Comment:
I thought the offset was supposed to be 0 for all parameters? I haven't gone through your IntegrationHeader code in detail yet so I am not sure but I thought you commented that somewhere.

Reply:
This offset is from the start of the parameters. The one you are thinking of (which is always 0) is the offset within a lambda capture object, which I had said it is not relevant because there is no lambda capture for a free function.